crates/pipeline-types/src/transport/nexmark.rs

```diff
@@ -0,0 +1,69 @@
+use enum_map::Enum;
+use serde::{Deserialize, Serialize};
+use utoipa::ToSchema;
+
+/// Configuration for generating Nexmark input data.
+///
+/// This connector must be used exactly three times in a pipeline if it is used
+/// at all, once for each [`NexmarkTable`].
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+pub struct NexmarkInputConfig {
+    /// Which table is this?
+    ///
+    /// Each table must appear in one connector.
+    pub table: NexmarkTable,
+
+    /// Overall behavior of the three linked input connectors.
+    ///
+    /// This may be specified only on one of the connectors and applies to all
+    /// them as a whole.
+    #[serde(default)]
+    pub options: Option<NexmarkInputOptions>,
+}
+
+/// Table in Nexmark.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema, Enum)]
+#[serde(rename_all = "snake_case")]
+pub enum NexmarkTable {
+    /// 92% of the events.
+    Bid,
+
+    /// 6% of the events.
+    Auction,
+
+    /// 2% of the events.
+    Person,
+}
+
+/// Configuration for generating Nexmark input data.
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+#[serde(default)]
+pub struct NexmarkInputOptions {
+    /// Number of events to generate.
+    pub events: u64,
+
+    /// Number of event generator threads.
+    pub threads: usize,
+
+    /// Number of events to generate and submit together.
+    pub batch_size: u64,
+
+    /// Whether to synchronize event generator threads after submitting each
+    /// batch.
+    ///
+    /// If true (which is the default), then the event generator threads will
+    /// submit data in lockstep, clustering their event sequence numbers. If
+    /// false, scheduling can cause some threads to get ahead of others.
+    pub synchronize_threads: bool,
+}
+
+impl Default for NexmarkInputOptions {
+    fn default() -> Self {
+        Self {
+            events: 100_000_000,
+            threads: 16,
```

Comment:
maybe this should use [available_paralleism](https://doc.rust-lang.org/stable/std/thread/fn.available_parallelism.html) to find a sensible default?

Reply:
I think that would give an answer that is too high on a system with lots of cores (I think it would answer 128 on my system).

The `run.py` script to drive this sets `threads` to the same as the number of workers. Maybe it should default to something small and let the script increase it.

Reply:
I was thinking of something like `max(4, available_parallelism / 8)` or similar
as with large core counts it might 16 might not be a good value on small core counts...

longer term might be better to just reuse our tokio runtime for all of this (since I think nexmark blocks anyways once it has buffered enough stuff..)